proxy/agent.go

```diff
@@ -129,25 +151,22 @@ func (a *agent) Run(stop <-chan struct{}) {
 	// High QPS is needed to process messages on all channels.
 	rateLimiter := flowcontrol.NewTokenBucketRateLimiter(float32(1), 10)
 
-	// Set default delay to a long duration - reconciliation is a no-op in the regular case.
-	// For permanent errors, reconciliation is still attempted once after the default delay.
-	defaultDelay := 1 * time.Hour
-	delay := defaultDelay
-
 	for {
 		rateLimiter.Accept()
 
 		select {
-		case config := <-a.config:
-			// reset retry budget if and only if the desired config changes
-			if !reflect.DeepEqual(a.desired, config) {
-				a.retryBudget = a.maxRetries
-				delay = defaultDelay
-				a.desired = config
+		case config := <-a.configCh:
+			if !reflect.DeepEqual(a.desiredConfig, config) {
```

Comment:
The client of the agent doesn't know if the new config "must" be sent. The way it is set up, config is generated every time the inputs change, and the config gets sent here. It is up to the agent to decide if it "must" be applied or not, and this check is doing that.